README.adoc

```diff
@@ -2185,6 +2185,28 @@ File.open('testfile') do |f|
 end
 ----
 
+=== Atomic File Operations [[atomic-file-operations]]
+
+When doing file operations after confirming the existence check of a file, frequent parallel file operations may cause problems that are difficult to reproduce.
+Therefore, it is preferable to use atomic file operations.
+
+[source,ruby]
+----
+# bad
+unless Dir.exist?(path)
+  FileUtils.mkdir(path)
+end
+
+if File.exist?(path)
+  FileUtils.remove(path)
+end
```

Comment:
It may be understood that those two snippets belong to a single example, while those are two separate ones and put corresponding good examples closer to bad ones.

WDYT of:
```ruby
# bad - race condition with another process may result in an error in `mkdir`
unless Dir.exist?(path)
  FileUtils.mkdir(path)
end

# good - atomic and idempotent creation
FileUtils.mkdir_p(path)

# bad - race condition with another process may result in an error in `remove`
if File.exist?(path)
  FileUtils.remove(path)
end

# good - atomic and idempotent removal
FileUtils.rm_f(path)
```

The above addresses two problems:
1. Confusion with what the example is
2. Order of good and bad

Leaving the order (bad, bad, good, good vs bad, good, bad, good) up to you. I found [one example](https://rubystyle.guide/#other-arg) of the second kind of ordering, but pretty sure one could find examples of bad, bad, good, good, too. So it's subjective and I leave it to your judgement how you want to do it.

Reply:
I indeed like the explanation of what concurrency issue can arise, maybe even specify the errors that could happen: `already exists` for `mkdir` and `does not exist` for `remove`?